### PR TITLE
Add a few Redis types

### PR DESCRIPTION
**Pull request by srittau**

### CODE REVIEW

**Comment by bochecha** [comment]
FWIW, this seems to work great at least in my use-case. :slightly_smiling_face:

File: third_party/2and3/redis/client.pyi [hunk 1]
```diff
@@ -31,16 +34,36 @@ def parse_hscan(response, **options): ...
...
+        encoding_errors: Text = ...,
+        charset: Optional[Text] = ...,
+        errors: Optional[Text] = ...,
+        decode_responses: bool = ...,
+        retry_on_timeout: bool = ...,
+        ssl: bool = ...,
+        ssl_keyfile: Optional[Text] = ...,
+        ssl_certfile: Optional[Text] = ...,
+        ssl_cert_reqs: Optional[Union[str, int]] = ...,
+        ssl_ca_certs: Optional[Text] = ...,
```
Comment (JelleZijlstra) [inline third_party/2and3/redis/client.pyi]: Missing the `max_connections=None` argument (https://github.com/andymccurdy/redis-py/blob/master/redis/client.py#L635).

File: third_party/2and3/redis/client.pyi [hunk 2]
```diff
@@ -117,7 +140,15 @@ class StrictRedis:
     def rename(self, src, dst): ...
     def renamenx(self, src, dst): ...
     def restore(self, name, ttl, value): ...
-    def set(self, name, value, ex=..., px=..., nx=..., xx=...): ...
+    def set(
+        self,
+        name: Text,
+        value: Any,
+        ex: Optional[int] = ...,
```
Comment (JelleZijlstra) [inline third_party/2and3/redis/client.pyi]: ex and px can also be timedeltas

**Comment by srittau** [comment]
@JelleZijlstra Thanks, done!

File: third_party/2and3/redis/client.pyi [hunk 3]
```diff
@@ -117,7 +141,15 @@ class StrictRedis:
     def rename(self, src, dst): ...
     def renamenx(self, src, dst): ...
     def restore(self, name, ttl, value): ...
-    def set(self, name, value, ex=..., px=..., nx=..., xx=...): ...
+    def set(
+        self,
+        name: Text,
```
Comment (joejuzl) [inline third_party/2and3/redis/client.pyi]: I think `name` (here and in the other methods too) can be either `str` or `bytes`.

**Pull request merged**